large-message-core/src/main/java/com/bakdata/kafka/AbstractLargeMessageConfig.java

```diff
@@ -260,9 +266,15 @@ private boolean enableAmazonS3PathStyleAccess() {
         final String roleExternalId = this.getString(S3_ROLE_EXTERNAL_ID_CONFIG);
         final String roleArn = this.getString(S3_ROLE_ARN_CONFIG);
         final String roleSessionName = this.getString(S3_ROLE_SESSION_NAME_CONFIG);
+        final String jwtPath = this.getString(S3_JWT_PATH_CONFIG);
 
         if (!isEmpty(roleExternalId) && !isEmpty(roleArn) && !isEmpty(roleSessionName)) {
-            final AWSCredentialsProvider provider =
+            final AWSCredentialsProvider provider = !isEmpty(jwtPath)?
```

Comment:
Please remove the check for empty `roleExternalId` from the outer if and create two ifs inside where you once check for empty `roleExternalId` and once for `jwtPath`